src/Target.h

```diff
@@ -50,6 +50,26 @@ struct Target {
     /** The bit-width of the target machine. Must be 0 for unknown, or 32 or 64. */
     int bits = 0;
 
+    /** The specific processor to be targeted, tuned for.
+     * Corresponds to processor_name_map in Target.cpp. */
+    enum Processor {
```

Comment:
Important Concern: Surely all these values need to echoed into an enum in HalideRuntime.h, just as we do for Features?

Nit #1: Add a comment reminding coders that these values are *not* going to have the same numeric values over time (as is the case for Features already)

Nit #2: Per previously suggestion, seems like it would be simpler to always list these lexicographically, to make keeping all future changes/additions consistent a simpler task. Is there a good reason to order them in a different way?

Reply:
> Important Concern: Surely all these values need to echoed into an enum in HalideRuntime.h, just as we do for Features?

Ok, done.
After that, this enum becomes sorted/sortable, but that new one can't be sorted.

Reply:
I'm not sure putting it in the runtime is necessary to land this.

Reply:
> I'm not sure putting it in the runtime is necessary to land this. Specifically, there is no use at runtime for them right now, though it could be surfaced somehow to help runtime dispatch. But that isn't plumbed through and can be added later.

And removed again :)